temporalio/worker/_nexus.py

```diff
@@ -0,0 +1,475 @@
+                    service=start_request.service,
+                    operation=start_request.operation,
+                    headers=header,
+                    request_id=start_request.request_id,
+                    callback_url=start_request.callback,
+                    inbound_links=[
+                        nexusrpc.handler.Link(url=l.url, type=l.type)
+                        for l in start_request.links
+                    ],
+                    callback_headers=dict(start_request.callback_header),
+                )
+                input = nexusrpc.handler.LazyValue(
+                    serializer=_DummyPayloadSerializer(
+                        data_converter=self._data_converter,
+                        payload=start_request.payload,
+                    ),
+                    headers={},
+                    stream=None,
+                )
+                try:
+                    result = await self._handler.start_operation(ctx, input)
+                except (
+                    nexusrpc.handler.UnknownServiceError,
+                    nexusrpc.handler.UnknownOperationError,
+                ) as err:
+                    # TODO(nexus-prerelease): error message
+                    raise nexusrpc.handler.HandlerError(
+                        "No matching operation handler",
+                        type=nexusrpc.handler.HandlerErrorType.NOT_FOUND,
+                        cause=err,
+                        retryable=False,
+                    ) from err
+
+            except nexusrpc.handler.OperationError as err:
+                return temporalio.bridge.proto.nexus.NexusTaskCompletion(
+                    task_token=task_token,
+                    completed=temporalio.api.nexus.v1.Response(
+                        start_operation=temporalio.api.nexus.v1.StartOperationResponse(
+                            operation_error=await self._operation_error_to_proto(err),
+                        ),
+                    ),
+                )
+            except BaseException as err:
+                handler_err = _exception_to_handler_error(err)
+                return temporalio.bridge.proto.nexus.NexusTaskCompletion(
+                    task_token=task_token,
+                    error=temporalio.api.nexus.v1.HandlerError(
+                        error_type=handler_err.type.value,
+                        failure=await self._exception_to_failure_proto(
+                            handler_err.__cause__
+                        ),
+                        retry_behavior=(
+                            temporalio.api.enums.v1.NexusHandlerErrorRetryBehavior.NEXUS_HANDLER_ERROR_RETRY_BEHAVIOR_RETRYABLE
+                            if handler_err.retryable
+                            else temporalio.api.enums.v1.NexusHandlerErrorRetryBehavior.NEXUS_HANDLER_ERROR_RETRY_BEHAVIOR_NON_RETRYABLE
+                        ),
+                    ),
+                )
+            else:
```

Comment:
I think all of this should be in the `try`. Users can throw exceptions out of their custom converters.

Reply:
I've refactored the Nexus worker substantially. Take another look at it to check that it's correct in this regard please!